From: Yuval Mintz <yuvalmin@broadcom.com>
Date: Mon, 12 Mar 2012 08:53:13 +0000
Subject: [PATCH 054/111] bnx2x: ethtool returns req. AN even when AN fails
Patch-mainline: v3.4-rc1
Git-commit: 241fb5d2b4205a523a50a85f9c312e1eda8ec53d
References: bnc#790597 FATE#313915

Previously, if autoneg failed, ethtool would return the achieved autoneg.
This patch corrects this, causing ethtool to return the requested autoneg
capabilities even if autoneg fails.

Signed-off-by: Yuval Mintz <yuvalmin@broadcom.com>
Signed-off-by: Eilon Greenstein <eilong@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_ethtool.c
@@ -1449,7 +1449,7 @@ static void bnx2x_get_pauseparam(struct
 			   BNX2X_FLOW_CTRL_AUTO);
 
 	if (!epause->autoneg)
-		cfg_reg = bp->link_vars.flow_ctrl;
+		cfg_reg = bp->link_params.req_flow_ctrl[cfg_idx];
 	else
 		cfg_reg = bp->link_params.req_fc_auto_adv;
 
